WordPress/src/main/java/org/wordpress/android/ui/jetpack/JetpackCapabilitiesUseCase.kt

```diff
@@ -31,58 +30,62 @@ class JetpackCapabilitiesUseCase @Inject constructor(
-                backup = capabilities.find { BACKUP_CAPABILITIES.contains(it) } != null
-        )
     }
 
-    suspend fun getOrFetchJetpackCapabilities(remoteSiteId: Long): List<JetpackCapability> {
-        return if (hasValidCache(remoteSiteId)) {
-            getCachedJetpackCapabilities(remoteSiteId)
-        } else {
-            fetchJetpackCapabilities(remoteSiteId)
-        }
-    }
+    fun getCachedJetpackPurchasedProducts(remoteSiteId: Long): JetpackPurchasedProducts =
+            mapToJetpackPurchasedProducts(getCachedJetpackCapabilities(remoteSiteId))
 
-    private suspend fun hasValidCache(remoteSiteId: Long): Boolean {
-        return withContext(bgDispatcher) {
-            val lastUpdated = appPrefsWrapper.getSiteJetpackCapabilitiesLastUpdated(remoteSiteId)
-            lastUpdated > currentDateProvider.currentDate().time - MAX_CACHE_VALIDITY
-        }
+    suspend fun fetchJetpackPurchasedProducts(remoteSiteId: Long): JetpackPurchasedProducts =
+            mapToJetpackPurchasedProducts(fetchJetpackCapabilities(remoteSiteId))
+
+    private fun mapToJetpackPurchasedProducts(capabilities: List<JetpackCapability>) =
+            JetpackPurchasedProducts(
+                    scan = capabilities.find { SCAN_CAPABILITIES.contains(it) } != null,
+                    backup = capabilities.find { BACKUP_CAPABILITIES.contains(it) } != null
+            )
+
+    fun hasValidCache(remoteSiteId: Long): Boolean {
+        val lastUpdated = appPrefsWrapper.getSiteJetpackCapabilitiesLastUpdated(remoteSiteId)
+        return lastUpdated > currentDateProvider.currentDate().time - MAX_CACHE_VALIDITY
     }
 
-    private suspend fun getCachedJetpackCapabilities(remoteSiteId: Long): List<JetpackCapability> {
-        return withContext(bgDispatcher) { appPrefsWrapper.getSiteJetpackCapabilities(remoteSiteId) }
+    private fun getCachedJetpackCapabilities(remoteSiteId: Long): List<JetpackCapability> {
+        return appPrefsWrapper.getSiteJetpackCapabilities(remoteSiteId)
     }
 
     private suspend fun fetchJetpackCapabilities(remoteSiteId: Long): List<JetpackCapability> {
-        return withContext(bgDispatcher) {
-            if (continuation != null) {
-                throw IllegalStateException("Request already in progress.")
-            }
-
-            dispatcher.register(this@JetpackCapabilitiesUseCase)
-            val response = suspendCoroutine<OnJetpackCapabilitiesFetched> { cont ->
-                val payload = FetchJetpackCapabilitiesPayload(remoteSiteId)
-                continuation = cont
-                dispatcher.dispatch(SiteActionBuilder.newFetchJetpackCapabilitiesAction(payload))
-            }
-
-            val capabilities: List<JetpackCapability> = response.capabilities ?: listOf()
-            if (!response.isError) {
-                updateCache(remoteSiteId, capabilities)
-            }
-            return@withContext capabilities
+        forceResumeDuplicateRequests(remoteSiteId)
```

Comment:
I've removed the "withContext" and replace throw exception with `forceResumeDuplicateRequests(remoteSiteId)`, the rest is without changes.